test/metabase/query_processor/middleware/expand_macros_test.clj

```diff
@@ -433,3 +433,23 @@
                                                            {:lib/uuid "7fb46618-622c-40f3-b0d4-4a779179f055"}
                                                            "0c586819-5288-4da9-adba-1ae904a34d5e"]]]}]
                 :database               (meta/id)}))))))
+
+(deftest ^:parallel mbql-2-macro-expansion-test
+  (testing "If the Macro is using super-legacy MBQL 2 can we still expand it?"
+    (is (=? {:aggregation [[:sum-where
+                            {}
+                            [:field {} (meta/id :venues :price)]
+                            [:<
+                             {}
+                             [:field {} (meta/id :venues :price)]
+                             4]]]}
+            (#'expand-macros/legacy-macro-definition->pMBQL
+             meta/metadata-provider
+             {:lib/type   :metadata/legacy-metric
+              :id         1
+              :name       "Metric 1"
+              :table-id   (meta/id :venues)
+              :definition {:source-table (meta/id :venues)
+                           :aggregation  [:sum-where
+                                          [:field (meta/id :venues :price) nil]
+                                          [:< [:field (meta/id :venues :price) nil] 4]]}})))))
```

Comment:
`:aggregation` is one aggregation rather than a sequence.

Reply:
This should probably be a comment in the code itself; that's a subtlety that would be easy to miss.